### ISSUE TITLE
Refactor UI+game to a single main loop

### ISSUE THREAD

**Issue by glebm** [diasurgical/devilutionX#3776]
If we ever want to port to a platform where we're not in control of the main loop, such as Emscripten, we need to refactor how we handle rendering and events first.

Currently, each "screen" in the menu has its own main loop, and the game has yet another main loop.

**Comment by Chance4us** [comment]
> If we ever want to port to a platform where we're not in control of the main loop, such as Emscripten, we need to refactor how we handle rendering and events first.
> 
> Currently, each "screen" in the menu has its own main loop, and the game has yet another main loop.

That's right,  unfortunately it is like that, and as far as I have seen there are several "rendering surface" confusions.

**Comment by AJenbo** [comment]
Here is a map of the menu flow:

![image](https://user-images.githubusercontent.com/204594/216949444-ce17df38-00b7-4216-87cc-95157876e1d2.png)

**Comment by glebm** [comment]
The hero screen can also have an error if there are more than the maximum number of characters

**Comment by AJenbo** [comment]
I'm starting this by restructuring the menus in to groups with sub panels.

Panels in a group will share the same background and loads/frees a common set of resources. Only when navigating from one group to another will assets be loaded/freed, this is also when the fade transition happens.
It's possible to jump to an arbitrary of any group (mainmenu -> hero/list -> hero/load -> multi/difficulty -> game) 

I'm going to try and wrangle the various properties that the menus interact with in a common place, rather then have them scattered around the game and various panels.

credits/-
error/-
mainmenu/-
ok/-
progress/-
title/-
yesno/-

hero/class
hero/create
hero/list
hero/load
hero/name

hub/chat
hub/create
hub/frinds
hub/join

multi/difficulty
multi/join
multi/list
multi/provider

options/groups
options/list
options/bind
options/enum

This probably won't get us directly to a well organized/structured system in one go, but I think it will at least be a step in the right direction
